packages/inference/src/providers/types.ts

```diff
@@ -0,0 +1,51 @@
+import type { InferenceTask } from "../types";
+
+/**
+ * Internal type for InferenceProvider definition.
+ *
+ * Not exported from the package
+ */
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace InferenceProviderTypes {
+	export type AuthMethod = "none" | "hf-token" | "credentials-include" | "provider-key";
+
+	export interface Config {
+		makeBaseUrl: MakeBaseUrl;
+		makeBody: MakeBody;
+		makeHeaders: MakeHeaders;
+		makeUrl: MakeUrl;
+		clientSideRoutingOnly?: boolean;
+	}
```

Comment:
A few changes:

- Made those types "internal" (not exported via the package), and namespaced them with `InferenceProviderTypes` to not conflict with `InferenceProvider` which is an enum/string union
- Directly created types for the makeX functions, instead of for the args. That way it's simpler to create new provider or edit return types.

I can revert those changes if you want cc @Wauplin @SBrandeis @hanouticelina 

(and just keep the `ProviderConfigWithOptionalModel` addition)